experimental/packages/web-common/src/SessionLogRecordProcessor.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Context } from '@opentelemetry/api';
+import { LogRecord, LogRecordProcessor } from '@opentelemetry/sdk-logs';
+import { ATTR_SESSION_ID } from '@opentelemetry/semantic-conventions/incubating';
+import { SessionProvider } from './types/SessionProvider';
+
+/**
+ * SessionLogRecordProcessor is a {@link SpanProcessor} adds the session.id attribute
+ */
+export class SessionLogRecordProcessor implements LogRecordProcessor {
+  private _sessionIdProvider: SessionProvider;
+
+  constructor(sessionIdProvider: SessionProvider) {
+    this._sessionIdProvider = sessionIdProvider;
+  }
+
+  onEmit(logRecord: LogRecord, _context?: Context | undefined): void {
+    const sessionId = this._sessionIdProvider.getSessionId();
```

Comment:
nit: technically, someone could pass in null as the provider, we probably should check that the provider is "present"